lib/crystalball/rails/map_generator/i18n_strategy/simple_patch.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module Crystalball
+  module Rails
+    class MapGenerator
+      class I18nStrategy
+        # Module to add new patched `load_file`, `store_translations` and `lookup`
+        # methods to `I18n::Backend::Simple`.
+        module SimplePatch
+          class << self
+            # Patches `I18n::Backend::Simple`.
+            def apply!
+              ::I18n::Backend::Simple.class_eval do
+                include SimplePatch
+
+                %i[load_file store_translations lookup].each do |method|
+                  alias_method :"cb_original_#{method}", method
+                  alias_method method, :"cb_patched_#{method}"
+                end
+              end
+            end
+
+            # Reverts original behavior of `I18n::Backend::Simple`
+            def revert!
+              ::I18n::Backend::Simple.class_eval do
+                %i[load_file store_translations lookup].each do |method|
+                  alias_method method, :"cb_original_#{method}"
+                  undef_method :"cb_patched_#{method}"
+                end
+              end
+            end
+          end
+
+          # Will replace original `I18n::Backend::Simple#load_file`.
+          # Stores filename in current thread
+          def cb_patched_load_file(filename)
+            Thread.current[:cb_locale_file_name] = filename
+            cb_original_load_file(filename)
+          end
+
+          # Will replace original `I18n::Backend::Simple#store_translations`.
+          # Adds filename for each value
+          def cb_patched_store_translations(locale, data, options = {})
```

Comment:
you can do `locale, data, *args` and be on a safe side if original interface changes